python/lsst/ci/build.py

```diff
@@ -345,6 +393,144 @@ def run(args):
 
         b = Builder(build_dir, manifest, progress, eups_obj)
         b.rm_status()
-        retcode = b.build()
-        b.write_status()
-        sys.exit(retcode == 0)
+
+        try:
+            if agent == "unknown_agent":
+                raise UnknownAgentError(agent)
+
+            # Verify PR info and post pending Github status
+            if pr_info.items:
+                post_statuses.post_all_status(
+                    pr_info, agent=agent, state="pending", description=f"Build started on {agent}"
+                )
+
+            # Attempt the build
+            retcode = b.build()
+
+        except UnknownAgentError as agent_err:
```

Comment:
This control flow is really confusing, especially doing a build in an `except` clause.  See above.